src/Symfony/Component/Security/Http/Authenticator/Passport/Badge/UserBadge.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\Passport\Badge;
+
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\EventListener\UserProviderListener;
+
+class UserBadge implements BadgeInterface
+{
+    private $username;
+    private $userLoader;
+    private $user;
+
+    public function __construct(?string $username, ?callable $userLoader = null)
+    {
+        $this->username = $username;
+        $this->userLoader = $userLoader;
+    }
+
+    public function getUsername(): string
+    {
+        return $this->username;
+    }
+
+    public function getUser(): UserInterface
+    {
+        if (null === $this->userLoader) {
+            throw new \LogicException(sprintf('No user loader is configured, did you forget to register the "%s" listener?', UserProviderListener::class));
+        }
+
+        $this->user = ($this->userLoader)($this->username);
+        if (null === $this->user) {
+            throw new UsernameNotFoundException();
+        }
+
+        return $this->user;
```

Comment:
We're setting the User into a `user` property but not really relying on that. If we called `getUser()` twice, it would execute our callback twice. I'm not sure which is the desired behavior :)